File: mysql-java-music/MusicApp.java
import java.sql.*;  
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Scanner;

/*
 
 javac MusicApp.java
 
 java -cp /home/codio/workspace/mysql-java-music/mysql-connector-java-8.0.13.jar:.: MusicApp

*/

     
public class MusicApp {  
    
    static Connection con;
    static Scanner sc;
    public static void main(String args[]){  
    
            sc = new Scanner(System.in);
            int option=6;
   
            try{  
  
                con=DriverManager.getConnection("jdbc:mysql://localhost/music?user=root");  
 
                Class.forName("com.mysql.cj.jdbc.Driver");  
 
                while (option != 5) {
      
    
                 System.out.println();
                 System.out.println("1. Add a musician");
                 System.out.println("2. Find a musician by id");
                 System.out.println("3. Find a musician by instrument");
                 System.out.println("4. Show all musicians");
                 System.out.println("5. Exit");
     
                 System.out.print("\nChoice : ");
        
                  option = sc.nextInt(); 
     
                 switch(option) {
        
                     case 2: findbyid();
                     break;
             
                     case 4: findall();
                     break;             
                 }
                        
                 }
                
            con.close();
        
            }catch(Exception e){ System.out.println(e);}
      }
    
public static void findall() {
    try{ 
        Statement stmt=con.createStatement();  
        ResultSet rs=stmt.executeQuery("select * from musicians");  
    
        while(rs.next()) {
        
            System.out.print(rs.getString("id") + " ");
            System.out.print(rs.getString("firstname") + " ");
            System.out.print(rs.getString("lastname") + " ");
            System.out.println(rs.getString("born"));
    
        }
    
          
    }catch(Exception e){ System.out.println(e);}
    
  }  
    
public static void findbyid() {
      
     System.out.print("Enter id: ");
     int id = sc.nextInt();
      
      try {
          
        PreparedStatement stmt=con.prepareStatement("select * from musicians where id = ?");   
        stmt.setInt(1,id);
        
        ResultSet rs=stmt.executeQuery();  
    
        while(rs.next()) {
        
            System.out.print(rs.getString("id") + " ");
            System.out.print(rs.getString("firstname") + " ");
            System.out.print(rs.getString("lastname") + " ");
            System.out.println(rs.getString("born"));
    
        }
    
          
    } catch(Exception e){ System.out.println(e);}
    
  }  
}

File: mysql-python-music/musicapp.py
import mysql.connector

# Execute:
# python3 musicapp.py
#

def findbyinst():
  inst = input("Instrument: ")

  mycursor = mydb.cursor()

  sql = "SELECT firstname, lastname, instrument " \
      "FROM musicians join instrumentsplayed " \
      "ON musicians.Id = instrumentsplayed.Musician " \
      "WHERE instrument = %s"
    
  arg = (inst, )

  mycursor.execute(sql, arg)

  myresult = mycursor.fetchall()

  for x in myresult:
    print(x)

def insert():
  fn = input("First Name: ")
  ln = input("Last Name: ")
  bornyear = input("Birth Year: ")

  mycursor = mydb.cursor()

  sql = "INSERT INTO musicians (firstname, lastname, born) " \
                                   "VALUES(%s,%s,%s)"
  arg = (fn, ln, bornyear, )

  mycursor.execute(sql, arg)

def showall():
     
  mycursor = mydb.cursor()

  mycursor.execute("SELECT * FROM musicians")

  myresult = mycursor.fetchall()

  for x in myresult:
    print(x)
    
  return;

def findbyid():
    
    id = input("Id: ")

    mycursor = mydb.cursor()

    sql = "SELECT * FROM musicians WHERE id = %s"
    arg = (id, )

    mycursor.execute(sql, arg)

    myresult = mycursor.fetchall()

    for x in myresult:
       print(x)
    
    return;

mydb = mysql.connector.connect(
  host="localhost",
  user="root",
  passwd="",
  database="music"
)

option = 0;

while option != 5: 
    
    print("")
    print("1. Add a musician")
    print("2. Find a musician by id")
    print("3. Find a musician by instrument")
    print("4. Show all musicians")
    print("5. Exit")
    
    option = int(input("Choice: "))
    print (option)
    if option == 1:
        insert()
    elif option == 4:
        showall()
    elif option == 2:
        findbyid()
    elif option == 3:
        findbyinst()


